bin/benchmark

```diff
@@ -178,3 +180,33 @@ longest_key_length = requests.keys.max_by(&:length).length
 
 puts "Benchmark results in seconds (slowest at top)\n\n"
 puts results.sort_by { |_, v| -v }.map { |k, v| "#{k.rjust(longest_key_length)} #{v}" }.join("\n")
+
+if File.exist?(CACHE_FILE_PATH)
+  main_results = JSON.parse(File.read(CACHE_FILE_PATH))
+  success = T.let(true, T::Boolean)
+
+  puts "\n\n"
+  puts "=" * 80
+  puts "Comparison with main branch:\n\n"
+  results.each do |method, current|
+    ratio = main_results[method].nil? ? 1.0 : 1.0 - (current / main_results[method])
+    absolute_ratio = ratio.abs
+    percentage = 100 * absolute_ratio
+    adjusted_method = method.rjust(longest_key_length)
+
+    if absolute_ratio < 0.15
+      puts "#{adjusted_method} unchanged"
+    elsif ratio.negative?
+      puts "#{adjusted_method} slower by #{percentage} %"
+    else
+      puts "#{adjusted_method} faster by #{percentage} %"
+    end
+  end
+
+  unless success
```

Comment:
This is always `true` since we don't write to it when we're slower. I'd think we'd want to fail the CI step so that we're notified when a particular PR introduces slowdown?